spec/pusher/push_notifications/generate_token_spec.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'jwt'
+
+RSpec.describe Pusher::PushNotifications::UseCases::GenerateToken do
+  subject(:use_case) { described_class.new(user: user) }
+
+  let(:user) { 'Elmo' }
+  let(:instance_id) { ENV['PUSHER_INSTANCE_ID'] }
+  let(:secret_key) { ENV['PUSHER_SECRET_KEY'] }
+
+  describe '#generate_token' do
+    subject(:generate_token) { use_case.generate_token }
+
+    context 'when user id is empty' do
+      let(:user) { '' }
+
+      it 'token generation will fail' do
+        expect { generate_token }.to raise_error(
+          Pusher::PushNotifications::UseCases::GenerateToken::GenerateTokenError
+        ).with_message(
+          'User Id cannot be empty.'
+        )
+      end
+    end
+
+    context 'when user id is too long' do
+      let(:user) { 'a' * 165 }
+
+      it 'user deletion request will fail' do
+        expect { generate_token }.to raise_error(
+          Pusher::PushNotifications::UseCases::GenerateToken::GenerateTokenError
+        ).with_message(
+          'User id length too long (expected fewer than 165 characters)'
+        )
+      end
+    end
+
+    context 'when user id is valid' do
+      it 'will generate the token' do
+        exp = Time.now.to_i + 24 * 60 * 60 # Current time + 24h
```

Comment:
I think this test might be flaky no? as it implies that this line and the line where it checks for the token runs in the same second.

I think I'd just check that the generated token is a valid JWT and that the `sub` is the user id.

Reply:
Fixed in 9316a09.